### DISCUSSION TITLE
What about Zig?

### DISCUSSION THREAD

**Discussion by FrancescoBorzi** [azerothcore/azerothcore-wotlk#19370] [Ideas]
I'm just curious to know if anyone has already looked into [Zig](https://ziglang.org/) and considered the possibility of leveraging its advantages for existing mature C++ projects like AzerothCore and similar.

In particular, this sounds quite attractive to me:

![image](https://github.com/azerothcore/azerothcore-wotlk/assets/75517/3954a2a0-f4ec-484c-b839-6b398221d024)

source: https://ziglang.org/

Especially the part of having a package manager and build system for existing C++ projects:

![image](https://github.com/azerothcore/azerothcore-wotlk/assets/75517/25b6d121-42b3-4a93-975f-ace156fa12f7)

source: https://ziglang.org/learn/why_zig_rust_d_cpp/

I did not have the chance to look into it in detail myself yet, but it sounds like something with great potential. Big projects like [Bun](https://bun.sh/) are already using it.

**Comment by FrancescoBorzi** [comment]
https://thenextweb.com/news/zig-highest-paying-programming-language

**Comment by Xunjin** [comment]
Never really used Zig, I do wonder how easy is to couple with C++ codebase and if there are all the necessary libraries that are used by the project.

I do believe that more projects with high performant and memory safe languages, like Zig and Rust, could give more traction/visibility to the project itself.

Perhaps some PoC on how adding Zig to the build system and some simple calls (some hardcoded log) would be a nice start.

**Reply by walkline** [reply]
I think the author intended to use Zig just as a replacement for our build system (cmake?). 
As for mixing C++ and Zig, I don't think it will happen because it's not very convenient.

**Reply by FrancescoBorzi** [reply]
indeed the first step would be using Zig as a build system
